### ISSUE TITLE
compress atomic vectors

### ISSUE THREAD

**Issue by krlmlr** [cynkra/constructive#138]
Allows devising a `rep()` call. Only if we have three or fewer distinct entries?

``` r
rle(rep(letters[3:1], 1:3))
#> Run Length Encoding
#>   lengths: int [1:3] 1 2 3
#>   values : chr [1:3] "c" "b" "a"
```

<sup>Created on 2023-04-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by moodymudskipper** [comment]
good idea, but then it makes sens to support a few more things:

* `rep(letters[3:1], 3)`
* `c(NA, NA)` -> `logical(2)` (+ other types)
* `seq(1, 7, 2)` integer and double, integer `i:j` is already supported by default.

I'm not sure if the number of distinct entries is the right parameter to trigger the behavior for `rep(letters[3:1], 1:3)`, I think the bottom line is : do we need less scalars to define the call ? `rep(c("c", "b", "a", 1:3)` is 6 scalars (if we ignore the 1:3 simplification, or just 5 if we don't), `c("c", "b", "b", "a", "a", "a")` is 6 too, so we're right at the edge here. 

So basically if `length(rle(x)$lengths) * 2 < length(x)` use `rep()`

**Comment by moodymudskipper** [comment]
Reopening because we're not robust to named vectors, let's have the compression work only on unnamed vectors it's the simple fix and I don't imagine use cases for named vectors.
